grails-app/services/org/bbop/apollo/ChadoHandlerService.groovy

```diff
@@ -1,49 +1,829 @@
+import org.gmod.chado.Cvterm
+
+import java.security.MessageDigest
+import java.sql.Timestamp
+
+/**
+ * Notes on Chado Export:
+ * Storing the sequence of chromosomes is optional with the default being true.
+ * Feature for chromosome do not have a corresponding featureloc as they serve as a srcfeature for other featurelocs.
+ * Symbol is treated as a synonym of a feature.
+ * A feature can have zero or more featurelocs where a typical feature will have one featureloc.
+ * featureloc coordinates are in interbase coordinate system.
+ * Orientation of a featureloc is represented by a value of 1 or -1 where default is 1.
+ * A featureloc is uniquely identified by the [feature_id, locgroup, rank].
+ * If a feature has more than one featureloc then it should be given a different locgroup and rank.
+ *
+ * Chado Compliance Layers
+ * Level 0: Relational schema - this basically means that the schema is adhered to
+ * Level 1: Ontologies - this means that all features in the feature table are of a type represented in SO and
+ * all feature relationships in feature_relationship table must be SO relationship types
+ * Level 2: Graph - all features relationships between a feature of type X and Y must correspond to relationship of
+ * that type in SO.
+ *
+ * 02.19.2016 - Initial version of export is aimed at exporting all information pertaining to Chado Sequence module
+ */
 
 @Transactional
 class ChadoHandlerService {
 
-    def writeFeatures(Organism organism,Collection<Feature> features){
+    def sequenceService
+    def featureRelationshipService
+    def transcriptService
+    def exonSerivce
+    def cdsService
+
+
+    HashMap<String, org.gmod.chado.Feature> existingChadoFeaturesMap = new HashMap<String, org.gmod.chado.Feature>()
+    HashMap<String, org.gmod.chado.Organism> chadoOrganismsMap = new HashMap<String, org.gmod.chado.Organism>()
+    HashMap<String, org.gmod.chado.Feature> chadoFeaturesMap = new HashMap<String, org.gmod.chado.Feature>()
+    HashMap<Sequence, org.gmod.chado.Feature> referenceSequenceMap = new HashMap<Sequence, org.gmod.chado.Feature>()
+    HashMap<String, org.gmod.chado.Dbxref> chadoDbxrefsMap = new HashMap<String, org.gmod.chado.Dbxref>()
+    HashMap<String, org.gmod.chado.Featureprop> chadoFeaturePropertiesMap = new HashMap<String, org.gmod.chado.Featureprop>()
+    HashMap<String, org.gmod.chado.FeaturePub> chadoFeaturePublications = new HashMap<String, org.gmod.chado.FeaturePub>()
+
+    HashMap<String, org.gmod.chado.Cvterm> cvtermsMap = new HashMap<String, org.gmod.chado.Cvterm>()
+    ArrayList<org.bbop.apollo.Feature> processedFeatures = new ArrayList<org.bbop.apollo.Feature>()
+    ArrayList<org.bbop.apollo.Feature> failedFeatures = new ArrayList<org.bbop.apollo.Feature>()
+
+    def writeFeatures(Organism organism, ArrayList<Feature> features) {
+        // Note: Eager fetch of features to avoid subsequent querying to database
+        def chadoFeatures = org.gmod.chado.Feature.all
```

Comment:
you probably want to select based on organism through Feature, FeatureLocation, Sequence, Organism.